lib/publish.js

```diff
@@ -150,7 +150,7 @@ function upload (pkg, isRetry, cached, opts) {
           )).finally(() => {
             // ignore errors.  Use the force.  Reach out with your feelings.
             return otplease(opts, opts => {
-              return upload(pkg, true, tarball, opts)
+              return upload(pkg, true, cached, opts)
```

Comment:
I think this should be replacing `upload` with `libpub`, not replacing `tarball` with `cached`.  Note that the function it's retrying is `libpub`, not `upload`?

We need a test in any event, and probably some more investigation.